From: Christian Melki <christian.melki@t2data.com>
Date: Tue, 17 May 2022 11:09:24 +0200
Subject: [PATCH] There is a potential infinite-loop in function
 _arc_error_normalized().

CVE: CVE-2019-6461
Upstream-Status: Pending
Signed-off-by: Ross Burton <ross.burton@intel.com>

The patch was imported from the Yocto project.

Signed-off-by: Christian Melki <christian.melki@t2data.com>
---
 src/cairo-arc.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/cairo-arc.c b/src/cairo-arc.c
index 390397bae104..f9249dbeb3c8 100644
--- a/src/cairo-arc.c
+++ b/src/cairo-arc.c
@@ -99,7 +99,7 @@ _arc_max_angle_for_tolerance_normalized (double tolerance)
     do {
 	angle = M_PI / i++;
 	error = _arc_error_normalized (angle);
-    } while (error > tolerance);
+    } while (error > tolerance && error > __DBL_EPSILON__);
 
     return angle;
 }
